app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -326,14 +326,28 @@ export class CloneRepository extends React.Component<
     this.updateAndValidatePath(newPath)
   }
 
-  private async doesPathExist(path: string) {
-    const exists = await pathExists(path)
-    // If the path changed while we were checking, we don't care anymore.
-    if (this.state.path !== path) {
-      return
-    }
+  private async validateEmptyFolder(path: string): Promise<null | Error> {
+    try {
+      const directoryFiles = await readdir(path)
+
+      // If the path has changed we no longer want the result
+      if (path !== this.state.path) {
+        return new Error('The path has been changed during check.')
+      }
+
+      if (directoryFiles.length === 0) {
+        return null
+      } else {
+        return new Error('The path folder contains files.')
+      }
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        // Folder does not exist
+        return null
```

Comment:
shouldn't this return an error so we don't consider it a success case in the caller?

Reply:
I have opted to return `null` here indicating a success, because if the folder does not exist then we should be able to clone to it. (Therefore, `ENOENT` indicates the only scenario in which we would be able to clone previous to these changes)

If you mean we should return an error and catch it in the caller, while still allow cloning when we catch this specific error, we could certainly do that. However, we would no longer be able to do the cleanup @shiftkey had suggested where we set the `error` state directly with the result of the validation function, though. That would be perfectly fine if that's what's preferable.

___

In regards to your input on the matter of when the path no longer matches the state, that's wonderful! I'm very glad to have a solution here so that we can press forward on this!
Thanks for the help on this. 

I will happily make the necessary changes once we are on the same page with your other comment.

Reply:
> I have opted to return null here indicating a success, because if the folder does not exist then we should be able to clone to it. (Therefore, ENOENT indicates the only scenario in which we would be able to clone previous to these changes)

Oh, you're totally right! Disregard my comment on that line. :)